gateway-service-api/src/main/proto/org/hypertrace/gateway/service/v1/baseline.proto

```diff
@@ -0,0 +1,33 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.hypertrace.gateway.service.v1.baseline";
+
+package org.hypertrace.gateway.service.v1.health;
+
+import "org/hypertrace/gateway/service/v1/gateway_query.proto";
+
+message BaselineEntity {
+  string id = 1;
+  string entity_type = 2;
+  map<string, org.hypertrace.gateway.service.v1.common.Value> attribute = 4;
+  map<string, org.hypertrace.gateway.service.v1.common.Baseline> baseline = 5;
+  map<string, org.hypertrace.gateway.service.v1.common.BaselineMetricSeries> baseline_metric_series = 6;
+}
+
+message BaselineEntitiesRequest {
+  string entity_type = 1;
+  sfixed64 start_time_millis = 2;
+  sfixed64 end_time_millis = 3;
+
+  // Filters using which the entities need to be filtered.
+  org.hypertrace.gateway.service.v1.common.Filter filter = 4;
```

Comment:
Last week, we discussed the problem with this approach - did something change?
To resummarize:
If we want to request baselines in parallel with the standard entity requests, we need to provide all of the inputs to fully specify a result set (sorts, limits, etc.) - given that we really don't want to duplicate that type of query processing (especially with the other changes around entity querying in flight - for example time agnostic queries, which @skjindal93 already landed) the option that likely makes more sense is to take in an explicit list of entity ids instead of full filters, and wait to query until after the first entity request comes back.

Reply:
I thought those changes will take time to finish, so i started with above approach. In the implementation i have just modified the request and call Entity Service, we can easily add sort, limits etc since implementation is same.  But more cleaner approach is the one you suggested, i agree with that. Will the same approach work for attributes as well ?

Reply:
I'm not sure what you mean - given the plan is to move baseline into an entirely separate service, the only scalable way for us to implement this is calling the entity API as is, then turning around with the returned result IDs in GQL and making a request for the baseline values for those IDs (which allows all paging/sorting/filtering to only be implemented in one spot). The attributes (as well as the paging/sorting/filtering) would stay on the first request, as they currently are, and the ID can then be used to correlate the two results and stitch them together.